src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -46,36 +46,28 @@ Copyright (c) .NET Foundation. All rights reserved.
   <!-- Hook for importing custom target framework parsing -->
   <Import Project="$(BeforeTargetFrameworkInferenceTargets)" Condition="$(BeforeTargetFrameworkInferenceTargets) != ''" />
 
-  <!-- Split $(TargetFramework) (e.g. net45) into short identifier and short version (e.g. 'net' and '45'). -->
-  <PropertyGroup Condition="'$(TargetFramework)' != '' and !$(TargetFramework.Contains(',')) and !$(TargetFramework.Contains('+'))">
-   <_ShortFrameworkIdentifier>$(TargetFramework.TrimEnd('.0123456789'))</_ShortFrameworkIdentifier>
-   <_ShortFrameworkVersion>$(TargetFramework.Substring($(_ShortFrameworkIdentifier.Length)))</_ShortFrameworkVersion>
-  </PropertyGroup>
-
-  <!-- Versions with dots are taken as is and just given leading 'v'. -->
-  <PropertyGroup Condition="'$(TargetFrameworkVersion)' == '' and '$(_ShortFrameworkVersion)' != '' and $(_ShortFrameworkVersion.Contains('.'))">
-    <TargetFrameworkVersion>v$(_ShortFrameworkVersion)</TargetFrameworkVersion>
-  </PropertyGroup>
+  <!--
+    Parse TargetFramework properties.
+  -->
+  <PropertyGroup Condition="'$(TargetFramework)' != '' and ('$(TargetFrameworkIdentifier)' == '' or '$(TargetFrameworkVersion)' == '')">
 
-  <!-- Versions with no dots and up to 3 characters get leading 'v' and implicit dots between characters. -->
-  <PropertyGroup Condition="'$(TargetFrameworkVersion)' == '' and '$(_ShortFrameworkVersion)' != ''">
-    <TargetFrameworkVersion Condition="$(_ShortFrameworkVersion.Length) == 1">v$(_ShortFrameworkVersion[0]).0</TargetFrameworkVersion>
-    <TargetFrameworkVersion Condition="$(_ShortFrameworkVersion.Length) == 2">v$(_ShortFrameworkVersion[0]).$(_ShortFrameworkVersion[1])</TargetFrameworkVersion>
-    <TargetFrameworkVersion Condition="$(_ShortFrameworkVersion.Length) == 3">v$(_ShortFrameworkVersion[0]).$(_ShortFrameworkVersion[1]).$(_ShortFrameworkVersion[2])</TargetFrameworkVersion>
+    <TargetFrameworkIdentifier>$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)'))</TargetFrameworkIdentifier>
+    <TargetFrameworkVersion>v$([MSBuild]::GetTargetFrameworkVersion('$(TargetFramework)', 3))</TargetFrameworkVersion>
```

Comment:
What does NuGet do when you the requested versionPartCount is less than the number of non-zero version parts?  @zkat @nkolev92 

I think maybe the correct behavior would be to interpret versionPartCount as a minimum, but not to truncate any real versions parts.  If that were the behavior, then we could just pass in 2 as the versionPartCount here and not worry about trimming a trailing `.0`.

In that case we wouldn'

Reply:
@dsplaisted did your comment get cut off? This seems like reasonable behavior although it would require a MSBuild change. I assume it's not blocking? I can make an issue to track updating to this logic.

Reply:
No, it's not blocking.  It looks like I had left in a fragment from before I reworded.  I've now deleted it.

Reply:
Great, I've made an issue for it: https://github.com/dotnet/sdk/issues/12614

Reply:
Pretty sure NuGet always returns an actual version, if nothing is provided, we default to 0.0.0, if it's an invalid version we fail.